src/Filemanager.tsx

```diff
@@ -176,37 +186,40 @@ function Filemanager() {
           setRenameFile(undefined);
 
           const body = {
-            dst: `${renameFile.path}/${newName}`,
+            dst: `${renameFile.nodeData!.path}/${newName}`,
           };
-          fetch(renameFile.url, {
-            method: "PATCH",
-            body: JSON.stringify(body),
-            headers: {
-              "Content-Type": "application/json;charset=utf-8",
-            },
-          })
-            .then((resp) => resp.json())
-            .then((data: api.Response) => {
-              if (data?.success) {
-                FilemanagerToaster.show({
-                  message: <div>"{renameFile.name}" has been renamed.</div>,
-                  className: "bp3-dark bp3-large bp3-text-large",
-                  timeout: 3000,
-                });
-              } else {
-                FilemanagerToaster.show({
-                  message: (
-                    <div>
-                      <p>Could not rename file "{renameFile.name}".</p>
-                      {data?.reason && <p>{data?.reason}</p>}
-                    </div>
-                  ),
-                  className: "bp3-dark bp3-large bp3-text-large",
-                  timeout: 3000,
-                  intent: Intent.DANGER,
-                });
-              }
-            });
+          const apiCall = mockApi
+            ? new Promise((resolve) => setTimeout(() => resolve({ success: true }), 1500))
+            : fetch(renameFile.nodeData!.url, {
+                method: "PATCH",
+                body: JSON.stringify(body),
+                headers: {
+                  "Content-Type": "application/json;charset=utf-8",
+                },
+              }).then((resp) => resp.json());
+          apiCall.then((data: any) => {
```

Comment:
```suggestion
              }).then((resp) => resp.json());
              
          apiCall.then((data: any) => {
```

Space it a bit. It might look clearer after